lib/annotate_rb/model_annotator/project_annotator.rb

```diff
@@ -41,7 +41,11 @@ def build_instructions_for_file(file)
         klass.reset_column_information
         annotation = Annotation::AnnotationBuilder.new(klass, @options).build
         model_name = klass.name.underscore
-        table_name = klass.table_name
+
+        # In multi-database configurations, it is possible for different models to have the same table name but live
+        #   in different databases. Here we are opting to use the table name to find related files only when the model
+        #   is using the primary connection.
+        table_name = klass.table_name if klass.connection_specification_name == ActiveRecord::Base.name
```

Comment:
The behavior of the connection_specification_name method differs in Rails versions prior to 6.0.x, but since Rails 6.0.x reached EOL on June 1st, 2023, I believe there's no need to support it.
- https://apidock.com/rails/v6.0.0/ActiveRecord/ConnectionHandling/connection_specification_name
- https://apidock.com/rails/v6.1.3.1/ActiveRecord/ConnectionHandling/connection_specification_name

Reply:
What happens if `klass.connection_specification_name != ActiveRecord::Base.name`, does `table_name` still need to get defined elsewhere?

Reply:
As long as you're following the naming conventions, I don't think it's necessary to define table_name elsewhere.
The reason is that it is used only for file lookup, and if table_name is nil during the lookup, a search based on model_name is also performed.

https://github.com/drwl/annotaterb/blob/df91a4e53836cd9569d3984c4170f5e4b6614dd5/lib/annotate_rb/model_annotator/file_name_resolver.rb#L16

Reply:
I will take another look this weekend, sorry for the delay. It seems like if we add this then the ActiveRecord version minimum will need to be bumped.